### PR TITLE
Update booking report form to require month and year

### PR DESCRIPTION
**Pull request by patrickjfl**
# Changes in this PR

This PR requires users to specify a month and year before downloading a booking report. Because of the existing method being used to pipe the response directly from the API, it is not practical to check for API errors, so we are validating the form input in the UI.

### Before

Screenshot 2023-05-10 at 11 13 41

### After

Screenshot 2023-05-10 at 11 12 47

Screenshot 2023-05-10 at 11 13 01

# Release checklist

[Release process documentation](https://dsdmoj.atlassian.net/wiki/spaces/AP/pages/edit-v2/4247847062?draftShareId=a1c360ab-bd31-4db1-aae3-7cc002761de9).

## Pre-merge checklist

- [ ] Are any changes required to dependent services for this change to work?
  (eg. CAS API)
- [ ] Have they been released to production already?

## Post-merge checklist

- [ ] [Manually approve](https://dsdmoj.atlassian.net/wiki/spaces/AP/pages/4247847062/Release+process#Manual-releases)
  release to test
- [ ] [Manually approve](https://dsdmoj.atlassian.net/wiki/spaces/AP/pages/4247847062/Release+process#Manual-releases)
  release to preprod
- [ ] [Manually approve](https://dsdmoj.atlassian.net/wiki/spaces/AP/pages/4247847062/Release+process#Manual-releases)
  release to prod

### CODE REVIEW

**Comment by patrickjfl** [comment]
[Change booking report endpoint to require a month and year](https://trello.com/c/7JMM3OR9/1058-change-booking-report-endpoint-to-require-a-month-and-year)

File: server/controllers/temporary-accommodation/manage/bookingReportsController.ts [hunk 1]
```diff
@@ -30,16 +33,25 @@ export default class BookingReportsController {
...
 
-        if (!probationRegionId) {
-          const error = new Error()
-          insertGenericError(error, 'probationRegionId', 'empty')
+        const error = new Error()
+
+        const fields = [{ probationRegionId }, { month }, { year }]
+
+        fields.forEach(field => {
+          const fieldName = Object.keys(field)[0]
```
Comment (richpjames) [inline server/controllers/temporary-accommodation/manage/bookingReportsController.ts]: I suggest we use more predictable keys in the objects in `fields`.  
This should help us for a few more reasons: 
- the type will be more suitable for inferrence 
- the `forEach` on 43 can be simplified 

If we instead use: 
``` const fields = [
          { name: 'probationRegionId', value: probationRegionId },
          { name: 'month', value: month },
          { name: 'year', value: year },
        ] 
``` 

then forEach can be:

        fields.forEach(field => {
          if (!field.value) {
            insertGenericError(error, field.name, 'empty')
          }
        })

File: server/controllers/temporary-accommodation/manage/bookingReportsController.ts [hunk 2]
```diff
@@ -30,16 +33,25 @@ export default class BookingReportsController {
...
+          if (!field[fieldName]) {
+            insertGenericError(error, fieldName, 'empty')
+          }
+        })
+
+        if (!probationRegionId || !month || !year) {
```
Comment (richpjames) [inline server/controllers/temporary-accommodation/manage/bookingReportsController.ts]: The conditional could be `if (fields.some(field => !field.value))`

**Review by richpjames — COMMENTED** [review]
I've left a couple of comments in the bookingReportsController that I think would improve things. But other than that looks good to me!

File: server/controllers/temporary-accommodation/manage/bookingReportsController.ts (hunk 1 above)
Comment (patrickjfl) [reply server/controllers/temporary-accommodation/manage/bookingReportsController.ts]: ah yeah, good shout!

**Review by richpjames — APPROVED** [review]

**Pull request merged**